test/master-playlist-controller.test.js

```diff
@@ -5718,3 +5718,163 @@ QUnit.test('false without nextPlaylist', function(assert) {
 
   this.env.log.warn.callCount = 0;
 });
+
+QUnit.module('MasterPlaylistController blacklistCurrentPlaylist', sharedHooks);
+
+QUnit.test('dont exclude only playlist unless it was excluded forever', function(assert) {
+  this.requests.length = 0;
+  this.player.dispose();
+  this.player = createPlayer();
+  this.player.src({
+    src: 'manifest/single.m3u8',
+    type: 'application/vnd.apple.mpegurl'
+  });
+
+  this.clock.tick(1);
+
+  this.masterPlaylistController = this.player.tech_.vhs.masterPlaylistController_;
+
+  // main
+  this.standardXHRResponse(this.requests.shift());
+  // media
+  this.standardXHRResponse(this.requests.shift());
+
+  let mpc = this.masterPlaylistController;
+  let mpl = mpc.masterPlaylistLoader_;
+  let playlist = mpl.master.playlists[0];
+  let shouldDelay = false;
+
+  mpl.load = (delay) => (shouldDelay = delay);
+
+  mpc.blacklistCurrentPlaylist();
+
+  assert.notOk('excludeUntil' in playlist, 'playlist was not excluded since excludeDuration was finite');
```

Comment:
Technically this is only saying `playlist not was not excluded`. I'm not sure if there's any check around the `blacklistDuration` here

Reply:
I guess I can verify that excludeUntil was set to a finite number.

Reply:
Oh, misread.
The blacklistDuration is either passed in on the error, the second argument, or is on the MPC. The default value is `300`. I think given that we expect the default value to be used, this should be good enough.